File: Ahorn/triggers/BossArenaTrigger.jl
module BossArenaTrigger

using ..Ahorn, Maple

@mapdef Trigger "MaggyHelper/BossArenaTrigger" BossArenaTrigger(x::Integer, y::Integer, bossEntityType::String="", bossName::String="Boss", createHealthUI::Bool=true, height::Integer=16, showHealthBar::Bool=true, startEncounter::Bool=false, triggerOnce::Bool=false, width::Integer=16)

const placements = Ahorn.PlacementDict(
    "Boss Arena Trigger" => Ahorn.EntityPlacement(BossArenaTrigger),
    "Boss Actor Encounter" => Ahorn.EntityPlacement(BossArenaTrigger)
)

function Ahorn.selection(entity::BossArenaTrigger)
    return Ahorn.getEntityRectangle(entity)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BossArenaTrigger, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.getEntityRectangle(entity), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/triggers/BossArenaTrigger.lua
local entity = {}

entity.name = "MaggyHelper/BossArenaTrigger"
entity.depth = 0
entity.placements = {
    {
        name = "Boss Arena Trigger",
        data = {
            width = 16,
            height = 16,
            bossName = "Boss",
            showHealthBar = true,
            createHealthUI = true,
            bossEntityType = "",
            startEncounter = false,
            triggerOnce = false
        }
    },
    {
        name = "Boss Actor Encounter",
        data = {
            width = 16,
            height = 16,
            bossName = "Boss",
            showHealthBar = true,
            createHealthUI = true,
            bossEntityType = "",
            startEncounter = true,
            triggerOnce = true
        }
    }
}

entity.fieldInformation = {
    bossName = {
        fieldType = "string"
    },
    bossEntityType = {
        fieldType = "string"
    }
}

entity.fieldOrder = {
    "x", "y", "width", "height",
    "bossName", "bossEntityType",
    "showHealthBar", "createHealthUI",
    "startEncounter", "triggerOnce"
}

return entity
